### PR TITLE
Port PDM to F4

### PR DESCRIPTION
**Pull request by gtaharaedmonds**
### Summary
Ported PDM to STM32F4 MCU.

### Changelist 
- PDM now uses STM32F4 MCU
- Changes PDM messages to JSON CAN
- Removed Io_AuxAux2 module because it is no longer relevant

### Testing Done
Flashed and debugged board, CAN TX is working.

### Checklist
*Please change `[ ]` to `[x]` when you are ready.*
- [x] I have read and followed the code conventions detailed in [README.md](../README.md) (*This will save time for both you and the reviewer!*).
- [x] If this pull request is longer then **500** lines, I have provided *explicit* justification in the summary above explaining why I *cannot* break this up into multiple pull requests (*Small PR's are faster and less painful for everyone involved!*).

### CODE REVIEW

File: can-bus/json/BMS/BMS_enum.json [hunk 1]
```diff
@@ -1,3 +1,6 @@
 {
-    
+    "AirState": {
```
Comment (will-chaba) [inline can-bus/json/BMS/BMS_enum.json:2]: I should keep this consistent on the BMS F4-Port PR

File: can-bus/json/BMS/BMS_tx.json [hunk 2]
```diff
@@ -14,5 +24,18 @@
...
+        "cycle_time": 1000,
+        "description": "Signals related to the contactors.",
+        "signals": {
+            "AirPositiveState": {
+                "enum": "AirState"
+            },
+            "AirNegativeState": {
+                "enum": "AirState"
+            }
+        }
```
Comment (will-chaba) [inline can-bus/json/BMS/BMS_tx.json:39]: Again, I should keep this consistent on my end. I like the name BMS_Contactors as opposed to air states. My msg-id for this is 112 tho. Idk, we could also just deal with the merge conflicts when they arise, I feel like they're inevitable.

**Review by will-chaba — COMMENTED** [review]
Looks good to me, we should co-ordinate to keep CAN msgs consistent for BMS

File: boards/PDM/Inc/main.h [hunk 3]
```diff
@@ -29,7 +29,7 @@ extern "C"
 #endif
 
 /* Includes ------------------------------------------------------------------*/
-#include "stm32f3xx_hal.h"
+#include "stm32f4xx_hal.h"
```
Comment (will-chaba) [inline boards/PDM/Inc/main.h:32]:
```suggestion
#include <Io_Hal.h>
```

File: boards/PDM/Src/Io/Io_SoftwareWatchdog.c [hunk 4]
```diff
@@ -1,6 +1,8 @@
-#include <stm32f3xx.h>
+#include <stm32f4xx.h>
```
Comment (will-chaba) [inline boards/PDM/Src/Io/Io_SoftwareWatchdog.c:1]:
```suggestion
#include "Io_Hal.h"
```

**Review by will-chaba — APPROVED** [review]

File: boards/PDM/Src/Io/Io_SoftwareWatchdog.c (hunk 4 above)
Comment (gtaharaedmonds) [reply boards/PDM/Src/Io/Io_SoftwareWatchdog.c:1]: good call

File: boards/PDM/Inc/main.h (hunk 3 above)
Comment (gtaharaedmonds) [reply boards/PDM/Inc/main.h:32]: this is autogenerated so not gonna edit here, but otherwise good call

**Pull request merged**